### PR TITLE
Integrate data field parser transaction and smart contract results 

### PR DESCRIPTION
**Pull request by miiu96**
- Integrate the data field parser in order to add new fields in the transaction API structure and smart contract results structure

### CODE REVIEW

File: testscommon/dataFieldParserStub.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package testscommon
+
+import datafield "github.com/ElrondNetwork/elrond-vm-common/parsers/dataField"
+
+type DataFieldParserStub struct {
```
Comment (bogdan-rosianu) [inline testscommon/dataFieldParserStub.go:5]: add comments to exported stuff

Comment (miiu96) [reply testscommon/dataFieldParserStub.go:6]: added

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/ElrondNetwork/elrond-go-core v1.1.16-0.20220523150518-f1519c41d352
 	github.com/ElrondNetwork/elrond-go-crypto v1.0.1
 	github.com/ElrondNetwork/elrond-go-logger v1.0.7
-	github.com/ElrondNetwork/elrond-vm-common v1.3.7-0.20220602122113-1f938c4d517a
+	github.com/ElrondNetwork/elrond-vm-common v1.3.7
```
Comment (iulianpascalau) [inline go.mod:15]: are we sure this is the correct commit?

Comment (miiu96) [reply go.mod:15]: Yes. I did a proper release after PR https://github.com/ElrondNetwork/elrond-vm-common/pull/92 was merged in the `master` branch

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
@@                            Log scanner                                  @@
##                            integrate-data-field-parser                  ##
================================================================================
- Known Warnings              21
- New Warnings                6
- Known Errors                0
- New Errors                  1
- Panics                      0
================================================================================

**Pull request merged**